src/dataMap/metaManager/lazyFactoryMap.js

```diff
@@ -0,0 +1,265 @@
+import { arrayFilter, arrayMap } from '../../helpers/array';
+import { assert, isFiniteUnsignedNumber, isNullish } from './utils';
+
+/**
+ * The LazyFactoryMap object holds key-value pairs in the structure similar to the
+ * regular Map. Once created, items can be moved around a grid depending on the operations
+ * performed on that grid - adding or removing rows. The collection requires "key"
+ * to be a zero-based index.
+ *
+ * It's essential to notice that the "key" index under which the item was created
+ * is volatile. After altering the grid, the "key" index can change.
+ *
+ * Having created N items with corresponding data
+ * +------+------+------+------+------+
+ * | 0/10 | 1/11 | 2/12 | 3/13 | 4/14 |  Keys (volatile zero-based index / internal storage index)
```

Comment:
Maybe information saying more precisely what these indexes are or that it's just an example with 10 items in the hole would do it yet more clear?

Reply:
I've added a little bit more description to this https://github.com/handsontable/handsontable/pull/6254/commits/eb1f3258945f22e65a1255460a62cc3da680e58e#diff-6401234415dbf15f9803715a2e1becc1R14